### PR TITLE
Loading design ports into pin planner

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
### Motivate of the pull request
Load design ports into pin planner.

### Describe the technical details
Run 'analyze' command if file with ports doesn't exists in the project folder. Then load ports into pin planner.

 ### Which part of the code base require a change
 
 - [x] Library: pinassignment
 - [ ] Plug-in: <Specify the plugin name>
 - [x] Engine
 - [ ] Documentation
 - [x] Regression tests
 - [ ] Continous Integration (CI) scripts

### screenshots
image

### CODE REVIEW

File: etc/templates/ports_test.json [hunk 1]
```diff
@@ -1,8 +1,10 @@
-{
-  "Top Module": "NAME_OF_THE_TOP_MODULE",
-  "Ports": [
-    {"name": "inp1", "direction": "Input", "range": {"msb":0, "lsb":0}, "type": "LOGIC"},
-    {"name": "out1", "direction": "Output", "range": {"msb":4, "lsb":0}, "type": "REG"},
-    {"name": "inout1", "direction": "Inout", "range": {"msb":2, "lsb":0}, "type": "LOGIC"}
-  ]
-}
+[
```
Comment (ravic-rs) [inline etc/templates/ports_test.json:1]: rename the file to ports_info.json, same as the upstream one for 3rd party users.

File: src/MainWindow/main_window.cpp [hunk 2]
```diff
@@ -657,6 +657,18 @@ void MainWindow::saveActionTriggered() {
 
 void MainWindow::pinAssignmentActionTriggered() {
   if (pinAssignmentAction->isChecked()) {
+    if (PinAssignmentCreator::searchPortsFile(
+            m_projectManager->getProjectPath())
+            .isEmpty()) {
+      auto res = Tcl_Eval(GlobalSession->TclInterp()->getInterp(), "analyze");
+      if (res != TCL_OK) {
+        QMessageBox::critical(this, "'analyze' command failed",
+                              "Please read console logs");
```
Comment (ravic-rs) [inline src/MainWindow/main_window.cpp:666]:
```suggestion
                              "Please read console logs for 'analyze' above");
```

File: tests/unittest/PinAssignment/PinAssignmentCreator_test.cpp [hunk 3]
```diff
@@ -46,3 +46,9 @@ TEST(PinAssignmentCreator, GetPortsWidget) {
   PinAssignmentCreator creator{new ProjectManager, new ToolContext("", "", "")};
   EXPECT_NE(creator.GetPortsWidget(), nullptr);
 }
+
+TEST(PinAssignmentCreator, SearchPortsFile) {
```
Comment (ravic-rs) [inline tests/unittest/PinAssignment/PinAssignmentCreator_test.cpp:50]:
```suggestion
TEST(PinAssignmentCreator, SearchPortsFileForEmptyFile) {
```

**Review by ravic-rs — COMMENTED** [review]
Couple of comments and suggestions to address.

File: etc/templates/ports_test.json (hunk 1 above)
Comment (KochynVolodymyr) [reply etc/templates/ports_test.json]: done

File: src/MainWindow/main_window.cpp (hunk 2 above)
Comment (KochynVolodymyr) [reply src/MainWindow/main_window.cpp]: done

File: tests/unittest/PinAssignment/PinAssignmentCreator_test.cpp (hunk 3 above)
Comment (KochynVolodymyr) [reply tests/unittest/PinAssignment/PinAssignmentCreator_test.cpp]: done

**Review by ravic-rs — APPROVED** [review]
Looks good.

**Pull request merged**